components/engine_traits/src/iterable.rs

```diff
@@ -117,8 +117,19 @@ pub trait RefIterable {
     fn iter(&self, opts: IterOptions) -> Result<Self::Iterator<'_>>;
 }
 
+pub trait IterMetricsCollector {
+    fn internal_delete_skipped_count(&self) -> usize;
```

Comment:
Should we include total scanned count as well? Otherwise we don't know the percentage of delete skipped or internal skipped.

Reply:
I will leave a todo and we should add these metrics when we need. This PR is to make the in-memory engine returns iteration related metrics like what did in RocksEngine which only include delete skipped count.